### PR TITLE
Replaced plugin with event listeners

### LINKED ISSUE #303: [Improvement]: Replace plugins with event listener
### Improvement description

https://github.com/pimcore/pimcore/pull/12633/files#diff-736172470d63f25a8d80e1457a5f8ef0ce9af5d798239430886fe1a410842aa3

Replace pimcore.plugin.broker.fireEvent with dispatchEvent and make events compatible for plugins like: 
https://github.com/pimcore/perspective-editor/pull/81

Change dependency from pimcore/pimcore to version 10.5

### PR DESCRIPTION
**Pull request by aashan10**
Resolves #303

### CODE REVIEW

**Review by kingjia90 — CHANGES_REQUESTED** [review]
The changes on `startup.js` might be missing
https://github.com/pimcore/customer-data-framework/blob/f028d4ce886814b9c1f8418c235fc08f710da38c/src/Resources/public/js/startup.js#L23

File: src/Resources/public/js/events.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+pimcore.events.cmf = {
```
Comment (dvesh3) [inline src/Resources/public/js/events.js]: please keep it consistent like other bundle event names so using something like `pimcore.events.cmfMenuReady`. please see https://github.com/pimcore/perspective-editor/blob/42675e3fb0b58e3f3ea33edd082434189de7a0f2/src/Resources/public/js/pimcore/perspective/events.js#L18

**Comment by dvesh3** [comment]
> The changes on `startup.js` might be missing
> 
> https://github.com/pimcore/customer-data-framework/blob/f028d4ce886814b9c1f8418c235fc08f710da38c/src/Resources/public/js/startup.js#L23

Please adapt the requested changes:
 1. dont extend from `pimcore.plugin.admin`
 2. remove `pimcore.plugin.broker.registerPlugin(this);` from initialize
 3. Add event listeners for startup.js methods e.g. `pimcoreReady`, `postOpenObject` etc.

File: src/Resources/public/js/events.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+pimcore.events.cmfMenuReady = 'cmfMenuReady';
```
Comment (dvesh3) [inline src/Resources/public/js/events.js]: please add bc layer like here https://github.com/pimcore/advanced-object-search/pull/169/files#diff-36b0cfa7350cf852204c156e1190840dfe7e9b086e4331c7dc9272f97babeefcR20

File: src/Resources/public/js/startup.js [hunk 3]
```diff
@@ -205,6 +210,12 @@ pimcore.plugin.customermanagementframework = Class.create(pimcore.plugin.admin,
         }
 
         this.addSegmentAssignmentTab(object, 'object', type);
+        document.dispatchEvent(new CustomEvent(pimcore.events.postOpenObject, {
```
Comment (dvesh3) [inline src/Resources/public/js/startup.js]: we should not re-dispatch events here. Instead add event listeners in initialize like this https://github.com/pimcore/advanced-object-search/pull/169/files#diff-109b95b1376bdd4064598c08caf2593a3472983b6b7c4c4951f8428b66fdc6cfR23

Comment (aashan10) [reply src/Resources/public/js/startup.js]: Should this be done for all the startup methods?

File: src/Resources/public/js/startup.js [hunk 4]
```diff
@@ -284,7 +296,9 @@ pimcore.plugin.customermanagementframework = Class.create(pimcore.plugin.admin,
 
         this.checkNewsletterQueueStatus(statusIcon);
     },
-    postOpenDocument: function (document, type) {
+    postOpenDocument: function (event) {
+        var document = event.detail.object;
```
Comment (dvesh3) [inline src/Resources/public/js/startup.js]:
```suggestion
        var document = event.detail.document;
```
https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/public/js/pimcore/events.js#L71

File: src/Resources/public/js/startup.js [hunk 5]
```diff
@@ -319,7 +333,9 @@ pimcore.plugin.customermanagementframework = Class.create(pimcore.plugin.admin,
         this.addSegmentAssignmentTab(document, 'document', type);
     },
 
-    postOpenAsset: function (asset, type) {
+    postOpenAsset: function (event) {
+        var asset = event.detail.object;
```
Comment (dvesh3) [inline src/Resources/public/js/startup.js]:
```suggestion
        var asset = event.detail.asset;
```
https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/public/js/pimcore/events.js#L35

File: src/Resources/public/js/startup.js (hunk 4 above)
Comment (blankse) [reply src/Resources/public/js/startup.js]: We should also bump `pimcore/pimcore` in composer.json to `^10.5.6` than.

Comment (dvesh3) [reply src/Resources/public/js/startup.js]: > We should also bump `pimcore/pimcore` in composer.json to `^10.5.6` than.

@aashan10 please incorporate this change as well. thanks!

File: src/Resources/public/js/startup.js [hunk 6]
```diff
@@ -193,10 +195,20 @@ pimcore.plugin.customermanagementframework = Class.create(pimcore.plugin.admin,
         }
 
         this.navEl.on('mousedown', toolbar.showSubMenu.bind(toolbar.cmfMenu));
-        pimcore.plugin.broker.fireEvent("cmfMenuReady", toolbar.cmfMenu);
+
```
Comment (dvesh3) [inline src/Resources/public/js/startup.js:198]: please fix params of pimcoreReady function

File: src/Resources/public/js/startup.js [hunk 7]
```diff
@@ -14,23 +14,25 @@
...
 
         pimcore.layout.toolbar.prototype.cmfMenu = this.menu;
+        document.addEventListener(pimcore.events.pimcoreReady, this.pimcoreReady.bind(this));
+        document.addEventListener(pimcore.events.postOpenDocument, this.postOpenDocument.bind(this));
+        document.addEventListener(pimcore.events.postOpenAsset, this.postOpenAsset.bind(this));
+        document.addEventListener(pimcore.events.postOpenObject, this.postOpenObject.bind(this));
     },
 
-    pimcoreReady: function (params, broker) {
+    pimcoreReady: function (viewport) {
```
Comment (dvesh3) [inline src/Resources/public/js/startup.js]:
```suggestion
    pimcoreReady: function (e) {
```

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**